server/mailet/integration-testing/src/test/java/org/apache/james/smtp/SmtpRandomStoringTest.java

```diff
@@ -0,0 +1,202 @@
+                .addMailet(MailetConfiguration.TO_BOUNCE));
+
+        jamesServer = TemporaryJamesServer.builder()
+            .withBase(MemoryJamesServerMain.SMTP_AND_IMAP_MODULE)
+            .withSmtpConfiguration(smtpConfiguration)
+            .withMailetContainer(mailetContainer)
+            .build(temporaryFolder);
+
+        MailboxProbeImpl mailboxes = jamesServer.getProbe(MailboxProbeImpl.class);
+        DataProbe dataProbe = jamesServer.getProbe(DataProbeImpl.class);
+        dataProbe.addDomain(DEFAULT_DOMAIN);
+        dataProbe.addUser(FROM, PASSWORD);
+        USERS.forEach(Throwing.<String>consumer(user -> {
+            dataProbe.addUser(user, PASSWORD);
+            mailboxes.createMailbox(MailboxPath.forUser(user, MailboxConstants.INBOX));
+        }).sneakyThrow());
+
+        imapProbe = jamesServer.getProbe(ImapGuiceProbe.class);
+    }
+
+    @After
+    public void tearDown() {
+        fakeSmtp.clean();
+        if (jamesServer != null) {
+            jamesServer.shutdown();
+        }
+    }
+
+    @Test
+    public void sendingTenMailsShouldBeRandomlyAssignedToEveryUsersINBOX() throws Exception {
+        int numberOfMails = 10;
+        createJamesServer(SmtpConfiguration.builder()
+            .requireAuthentication()
+            .withAutorizedAddresses("172.0.0.0/8"));
+
+        SMTPMessageSender authenticatedSmtpConnexion = messageSender.connect(LOCALHOST_IP, jamesServer.getProbe(SmtpGuiceProbe.class).getSmtpPort())
+                .authenticate(FROM, PASSWORD);
+
+        IntStream.range(1, numberOfMails)
+            .forEach(Throwing.intConsumer(index ->
+                authenticatedSmtpConnexion
+                    .sendMessage(buildMail("Message " + index))).sneakyThrow());
+
+        Collection<IMAPMessageReader> connexions = USERS
+                .stream()
+                .map(Throwing.function(username -> new IMAPMessageReader()
+                    .connect(LOCALHOST_IP, imapProbe.getImapPort())
+                    .login(username, PASSWORD)))
+                .collect(Guavate.toImmutableList());
+
+        awaitAtMostTenSeconds
+            .untilAsserted(() -> checkMailboxes(connexions));
+    }
+
+    private void checkMailboxes(Collection<IMAPMessageReader> connexions) {
+        assertThat(connexions
+            .stream()
+            .mapToLong(this::checkMailbox)
+            .sum())
+            .isBetween(USERS_NUMBERS * 4L, USERS_NUMBERS * 8L);
+    }
+
+    private Long checkMailbox(IMAPMessageReader imapMessageReader) {
+        try {
+            long numberOfMails = imapMessageReader
+                .getMessageCount(IMAPMessageReader.INBOX);
+
+            assertThat(numberOfMails)
+                    .isGreaterThan(0);
+
+            return numberOfMails;
+        } catch (IOException e) {
+            return 0L;
+        }
+    }
+
+    private Mail buildMail(String subject) throws MessagingException {
```

Comment:
could you transform the exception into a runtime one in this method to make the calling site less bloated?

Reply:
`sendMessage` also requires it